feat(datasource/custom): add `html` format (#24403)
renovatebot/renovate: lib/modules/datasource/custom/readme.md

After the conversion, any `jsonata` rules defined in the `transformTemplates` section will be applied as usual to further process the JSON data.

#### HTML

If the format is set to `html`, Renovate will call the HTTP endpoint with the `Accept` header value `text/html`.
The body of the response will be treated as a HTML document, and all hyperlinks will be converted to versions.

For the following HTML document:

```html
<html>
  <body>
    <a href="package-1.0.tar.gz">package-1.0.tar.gz</a>
    <a href="package-2.0.tar.gz">package-2.0.tar.gz</a>
  </body>
</html>
```

The following JSON will be generated:

```json
{
  "releases": [
    {
      "version": "package-1.0.tar.gz",
      "sourceUrl": "https://example.com/package-1.0.tar.gz"
    },
    {
      "version": "package-1.0.tar.gz",
      "sourceUrl": "https://example.com/package-2.0.tar.gz"
    }
  ]
}
```

> `href` attribute value becomes the `version`, and `sourceUrl` is set to the full resolved URL.

Author: Removed the sentence completely, because there's no `sourceUrl` anymore

> To extract the actual version number, you could use `extractVersion` or JSONata rules.

Reviewer: ```suggestion
To extract the version number, you may use [`extractVersion`](https://docs.renovatebot.com/configuration-options/#extractversion) or JSONata rules.
```

I think the only versions are _actual_ version numbers, so we can drop the word actual.

We should also link to the `extractVersion` documentation. I'd prefer a relative link here.

After the conversion, any `jsonata` rules defined in the `transformTemplates` section will be applied as usual to further process the JSON data.

#### HTML

If the format is set to `html`, Renovate will call the HTTP endpoint with the `Accept` header value `text/html`.
The body of the response will be treated as a HTML document, and all hyperlinks will be converted to versions.

For the following HTML document:

```html
<html>
  <body>
    <a href="package-1.0.tar.gz">package-1.0.tar.gz</a>
    <a href="package-2.0.tar.gz">package-2.0.tar.gz</a>
  </body>
</html>
```

The following JSON will be generated:

```json
{
  "releases": [
    {
      "version": "package-1.0.tar.gz",
    },
    {
      "version": "package-1.0.tar.gz",
    }
  ]
}
```

> After the conversion, any `jsonata` rules defined in the `transformTemplates` section will be applied as usual to further process the JSON data.

Reviewer: ```suggestion
After the conversion, any `jsonata` rules defined in the `transformTemplates` section will be applied to process the JSON data.
```

  ]
}
```

### nginx directory listing

Sometimes all you have is a directory with files, and a HTTP server that can generate directory listings.

Let's use nginx itself as an example:

```json
{
  "customDatasources": {
    "nginx": {

>       "defaultRegistryUrlTemplate": "http://nginx.org/download",

Reviewer: Is nginx still using HTTP (without the S)? I'd prefer if we could show a HTTPS example, if at all possible.

      "format": "html"
    }
  },
  "packageRules": [
    {
      "matchDatasources": ["custom.nginx"],
      "extractVersion": "^nginx-(?<version>.+)\\.tar\\.gz$"
    }
  ]
}
```

### HTML page

> `html` format can also be used to extract versions from a typical "Downloads" page:

Reviewer: ```suggestion
You can use the `html` format to extract versions from a typical "Downloads" page:
```
